### PR TITLE
[Debugger] Can't edit function breakpoint

### PR DESCRIPTION
**Pull request by iainx**
Enable editing of breakpoint locations, either line, function or exception

Fixes VSTS #528267

### CODE REVIEW

File: main/src/addins/MonoDevelop.Debugger/MonoDevelop.Debugger/BreakpointPropertiesDialog.cs [hunk 1]
```diff
@@ -357,6 +357,12 @@ void SaveFunctionBreakpoint (FunctionBreakpoint fb)
 			fb.ParamTypes = parsedParamTypes;
 		}
 
+		void SaveCatchpoint (Catchpoint cp)
+		{
+			cp.ExceptionName = entryExceptionType.Text;
+			cp.IncludeSubclasses = checkIncludeSubclass.Active;
```
Comment (slluis) [inline main/src/addins/MonoDevelop.Debugger/MonoDevelop.Debugger/BreakpointPropertiesDialog.cs]: Will this work in the middle of a debugging session? will the breakpoing be re-bound?

Comment (iainx) [reply main/src/addins/MonoDevelop.Debugger/MonoDevelop.Debugger/BreakpointPropertiesDialog.cs]: Yes, it appears to in my tests.

**Comment by DavidKarlas** [comment]
@iainx You probably did changes also in debugger-libs repo that you need to push/open PR for.

**Comment by iainx** [comment]
@DavidKarlas yes, you're right, I didn't realise I was editing in there as well

PR filed: https://github.com/mono/debugger-libs/pull/175

**Comment by Therzok** [comment]
I think this needs to be rebased on master, otherwise VSTs will just keep building this commit.

**Comment by DavidKarlas** [comment]
@iainx I tried if re-bounding works, doesn't seem to work in my case:
1) Add function breakpoint to `MainClass.test1()`
2) Start debugging
3) `test1` method is hit
4) Edit function breakpoint and change it to `MainClass.test2()`
5) Continue execution
6) It should break when enters `test2` method, but it doesn't it continues execution and finished process
```
class MainClass
{
    public static void Main(string[] args)
    {
        test1();
    }

    private static void test1()
    {
        test2();
    }

    private static void test2()
    {
        Console.WriteLine("test2");
    }
}
```

**Comment by iainx** [comment]
@DavidKarlas Yes, you're right, I only tested moving the breakpoints when the code wasn't running, and looking at the soft debugger, the code to move breakpoints when the debugger is running hasn't been implemented, and it beyond my expertise at the moment (is it even supported in Mono?)

So, I've made it so that the breakpoints are only editable when the debugger session isn't running, which improves the situation slightly at least

**Comment by ketttY** [comment]
@monojenkins rebase

**Comment by ketttY** [comment]
@monojenkins rebase

**Comment by ketttY** [comment]
@monojenkins rebase

**Comment by ketttY** [comment]
@monojenkins rebase

**Comment by sgmunn** [comment]
@monojenkins rebase

**Comment by ketttY** [comment]
Current version:
EditFunctionBreakpoint1080

**Comment by ketttY** [comment]
Test Case: https://devdiv.visualstudio.com/DevDiv/_workitems/edit/1052373

**Comment by ketttY** [comment]
@monojenkins rebase

**Comment by mrward** [comment]
Compat check error with Xamarin.Mac is known and unrelated to this change.

**Pull request merged**